src/libp2p_transport_tcp.erl

```diff
@@ -55,6 +54,20 @@ dial(MAddr) ->
         {error, Reason} -> {error, Reason}
     end.
 
+tcp_listen_addrs(Socket) ->
+    {ok, SockAddr={IP, Port}} = inet:sockname(Socket),
+    case lists:all(fun(D) -> D == 0 end, tuple_to_list(IP)) of
+        false ->
+            [multiaddr(SockAddr)];
+        true ->
+            % all 0 address, collect all non loopback interface addresses
+            {ok, IFAddrs} = inet:getifaddrs(),
+            [multiaddr({Addr, Port}) ||
+                {_, Opts} <- IFAddrs, {addr, Addr} <- Opts, {flags, Flags} <- Opts,
+                size(Addr) == size(IP),
```

Comment:
Won't this filter out ipv6? we shoyld only filter out ipv6 addresses that begin with 16#fe80 (or `65152` decimal).

Reply:
Oh, I see now, there's no way to bind to both ipv4 and v6 at the same time.